app/js/handlebarsScoreTable.ts

```diff
@@ -283,12 +283,30 @@ async function addEventListenersForDownloadButtons() {
     document.querySelectorAll('.download-user-results-button').forEach((button) => {
         button.addEventListener("click", (e: any) => {
             e.preventDefault();
-            getData("result?id=" + e.target.parentElement.getAttribute("dataId")).then(resultData => {
+            getData("result").then(resultData => {
+                let resultsObj = {data:[]}
+                let userName = ''
+                let userPercentage = ''
+
+                resultData.data.forEach(obj => {
+                    if(obj.id == e.target.getAttribute("dataId")){
+                        resultsObj.data.unshift(obj)
+                    }
+                })
+                let resultNumber = e.target.getAttribute('datanumber')
+                let individualResult = {success: true, message: "Successfully retrieved results.",data:{}}
+                individualResult.data = resultsObj.data[resultNumber]
+                let moreInfoButtonsData = document.querySelectorAll('button.more-info-button')
+                moreInfoButtonsData.forEach(buttonWithData => {
+                    if(buttonWithData.getAttribute('dataid') == individualResult.data.id){
+                        userName = buttonWithData.getAttribute('dataname')
+                    }
+                })
```

Comment:
instead of looping over all the more info buttons to find the one with the right `dataid` attribute, you can use a CSS attribute selector, see here:

https://www.w3schools.com/css/css_attribute_selectors.asp